docs/modular-resources/advanced/mlmodel-design.md

```diff
@@ -0,0 +1,120 @@
+---
+title: "Design your ML Models for Vision"
+linkTitle: "Vision Service and Modular ML Model Design"
+weight: 60
+type: "docs"
+tags: ["data management", "ml", "model training"]
+description: "Design your ML Model service to work with Viam's vision services."
+icon: "/services/icons/ml.svg"
+# SME: Bijan Haney
+---
+
+Models of Viam's [Machine Learning (ML) model service](/services/ml/) allow you to deploy machine learning models to your smart machine.
+Vision services, like [an `"mlmodel"` detector](/services/vision/detection/#configure-an-mlmodel-detector) or [classifier](/services/vision/classification/#configure-an-mlmodel-classifier), enable you to identify and classify objects in images with the predictions those deployed models make.
+
+The two services work closely together, with the vision service relying on the deployed ML model to make inferences.
+If [designing your own ML Model service](/modular-resources/) to add to [the Registry](https://app.viam.com/registry), you must try to make your ML models' shapes match the input and output tensors the vision service expects to work with if you want the two services to coordinate in classification or detection.
+
+To know what to expect, the vision service looks for descriptions of these characteristics in the [metadata](/services/ml/#metadata) of the ML model, as defined in [the Python SDK](https://python.viam.dev/autoapi/viam/gen/service/mlmodel/v1/mlmodel_pb2/index.html#viam.gen.service.mlmodel.v1.mlmodel_pb2.Metadata).
+For an example of this, see [Example Metadata](#example-metadata).
+
+## Input tensor: `input_info` in metadata
+
+For both [classification](/services/vision/classification/) and [detection](/services/vision/detection/) models, the vision service sends an input tensor to the ML Model with the following structure:
+
+- One input tensor called `"image"` with type `uint8` or `float32` and shape `(1, height, width, 3)`, with the last channel, `3` being the RGB bytes of the pixel.
+- If image `height` and `width` are unknown or variable, then `height` and/or `width` `= -1`. During inference runtime the image will have a known height and width.
+
+## Output tensors: `output_info` in metadata
+
+Data can be returned by the ML model in many ways, due to the variety of machine learning models for computer vision.
+While the vision service tries to take into account many different forms of models by looking at the metadata of the model, if the model does not provide metadata, the vision service will make guesses.
+If you need to add structure and metadata, output that is organized in terms of the "ideal guess" will work out of the box.
+The ideal guesses of the vision service are as follows:
+
+For labels:
+
```

Comment:
Can we add this in front of what's below: "Many computer vision models have an associated 'labelfile.txt' that simply lists the class labels associated with the model."

Reply:
Yep! Added